Monika After Story/game/screens.rpy

```diff
@@ -1421,6 +1421,7 @@ screen preferences():
                 style_prefix "slider"
                 box_wrap True
 
+                # TODO: clean this up, we should not exucute py code in screens
```

Comment:
what's your idea for this TODO? The pycode here needs to run with the screen because it handles updating screen text and keeping the suntime sliders in a valid order.

Reply:
I'll add validators which would be run on the sliders changing. Alternatively validate the values are in the valid range on the menu exit.

For generating labels text we can wrap that in functions. Like
```renpy
# Say for sunset
label _("[[ " + mas_suntime.get_ss_disp() + " ]")
```

Screens have an odd behavior and executing py code like that isn't ideal.

Reply:
ok, but validating on exit is a no-go. The sliders should validate immediately otherwise users can drag them to be invalid and its an ugly experience.